src/util/locking/RedisLocker.ts

```diff
@@ -53,8 +62,12 @@ export class RedisLocker implements ReadWriteLocker, ResourceLocker, Initializab
   private readonly attemptSettings: Required<AttemptSettings>;
   private finalized = false;
 
-  public constructor(redisClient = '127.0.0.1:6379', attemptSettings: AttemptSettings = {}) {
-    this.redis = this.createRedisClient(redisClient);
+  public constructor(
+    redisClient = '127.0.0.1:6379',
+    attemptSettings: AttemptSettings = {},
+    redisOptions: RedisOptions = {},
+  ) {
```

Comment:
(see previous interface change I requested): This would then become 
```ts
public constructor(
    redisClient = '127.0.0.1:6379',
    attemptSettings: AttemptSettings = {},
    redisSettings: RedisSettings = { namespacePrefix: '' }
}
```